src/test/java/io/confluent/connect/jdbc/source/JdbcSourceTaskUpdateTest.java

```diff
@@ -316,11 +369,39 @@ public void testManualIncrementingRestoreOffset() throws Exception {
     PowerMock.verifyAll();
   }
 
+  @Test
+  public void testAutoincrementRestoreNoVersionOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(null, 1L);
+    testAutoincrementRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap())
+    );
+  }
+
+  @Test
+  public void testAutoincrementRestoreVersionOneOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(null, 1L);
+    testAutoincrementRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap())
+    );
+  }
+
   @Test
   public void testAutoincrementRestoreOffset() throws Exception {
+    TimestampIncrementingOffset oldOffset = new TimestampIncrementingOffset(null, 0L);
     TimestampIncrementingOffset offset = new TimestampIncrementingOffset(null, 1L);
-    expectInitialize(Arrays.asList(SINGLE_TABLE_PARTITION),
-                     Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap()));
+    Map<Map<String, String>, Map<String, Object>> offsets = new HashMap<>();
+    offsets.put(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap());
+    offsets.put(SINGLE_TABLE_PARTITION, oldOffset.toMap());
+    testAutoincrementRestoreOffset(offsets);
```

Comment:
How about a comment here to say that we want to always use the offset with the latest protocol found?